### PR TITLE
[v5] breaking: make use-sync-external-store an optional peer dependency

### PR DESCRIPTION
**Pull request by charkour**
## Summary

- for #2138
- waiting on #2236

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -251,22 +248,27 @@
...
     "shx": "^0.3.4",
     "typescript": "^5.3.2",
+    "use-sync-external-store": "^1.2.0",
     "vitest": "^0.34.6"
   },
   "peerDependencies": {
-    "immer": ">=9.0",
     "@types/react": ">=16.8",
-    "react": ">=16.8"
+    "immer": ">=9.0",
```
Comment (charkour) [inline package.json:256]: Yarn prefers to have these dependencies listed in alphabetical order which is why these changed.

File: package.json [hunk 2]
```diff
@@ -251,22 +248,27 @@
...
     },
-    "@types/react": {
+    "immer": {
       "optional": true
     },
     "react": {
       "optional": true
+    },
+    "use-sync-external-store": {
+      "optional": true
```
Comment (charkour) [inline package.json:271]: I opted to make this optional because it is only used with `react` which is also optional.

Comment (dai-shi) [reply package.json:256]: Yes, that was my intention. I forgot react was already optional.

**Review by dai-shi — APPROVED** [review]
LGTM

**Comment by dai-shi** [comment]
Oooops, we forgot to remove `use-sync-external-store` from react.ts.

**Pull request merged**